src/App.js

```diff
@@ -1,41 +1,51 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
+import { createBrowserRouter, RouterProvider, Outlet } from 'react-router-dom';
+
 import Header from './components/Header';
 import Body from './components/Body';
 import Footer from './components/Footer';
-
+import About from './components/About';
+import Contact from './components/Contact';
+import RestuarantMenu from './components/RestuarantMenu';
+import Error from './components/Error';
 
 const App = () => {
     return(
         <div className="app-container">
             <Header />
-            <Body />
+            <Outlet />
             <Footer />
         </div>        
     )
 }
 
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+const appRouter = createBrowserRouter([
+    {
+        path: '/',
+        element: <App />,
+        children: [
+            {
+                path: '/',
+                element: <Body />
+            },
+            {
+                path: '/about',
+                element: <About />
+            },
+            {
+                path: '/contact',
+                element: <Contact />
+            },
+            {
+                path: '/restuarant/:resId',
+                element: <RestuarantMenu />
+            }
+        ],
+        errorElement: <Error />
+    }
+])
 
 const rootElement = document.getElementById("root");
 const root = ReactDOM.createRoot(rootElement);
-root.render(<App/>);
+root.render(<RouterProvider router={appRouter} />);
```

Comment:
Implementation of routing by creating Routing configuration with createBrowserRouter 
and providing it to app using RouterProvider